contribs/gnostats/agent/collector_test.go

```diff
@@ -0,0 +1,370 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	ctypes "github.com/gnolang/gno/tm2/pkg/bft/rpc/core/types"
+	"github.com/gnolang/gno/tm2/pkg/bft/state"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/p2p"
+	"github.com/gnolang/gnostats/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// MockRPCClient is a mock of the rpcClient used for testing
+type MockRPCClient struct {
+	mock.Mock
```

Comment:
I'm gonna be honest, I'm not a huge fan of using testify's mock functionality -- it brings back not-so-fond memories of JS testing 🤷‍♂️

Having said this, I think we should keep it regardless, it's cool to have multiple ways to test things

Reply:
Okay I wont use it next time 😉